### PR TITLE
Upgrading redux-offline and netinfo dependencies for react native

### PR DESCRIPTION
**Pull request by Ashish-Nanda**
*Issue #, if available:*

*Description of changes:*
Upgrading redux-offline and netinfo dependencies for react native to use latest versions. This is because newer versions of NetInfo deprecated some APIs that were used by the version of redux-offline we were using.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ucallQA** [comment]
is there an ETA when this will be merged?

**Comment by DrWorkhard** [comment]
Waiting for this as well, if I understand correctly this makes amplify compatible with expo 37

**Comment by Ashish-Nanda** [comment]
@DrWorkhard we were waiting for @react-native-community/async-storage to be supported in expo because redux-offline uses has a dependency on it, and and this change would not work in Expo without it.

This just happened over the weekend: https://github.com/react-native-community/async-storage/pull/368#issuecomment-632960787

We will probably test the change in both bare and expo managed apps and then merge the PR later this week.

**Comment by DrWorkhard** [comment]
> 
> 
> @DrWorkhard we were waiting for @react-native-community/async-storage to be supported in expo because redux-offline uses has a dependency on it, and and this change would not work in Expo without it.
> 
> This just happened over the weekend: [react-native-community/async-storage#368 (comment)](https://github.com/react-native-community/async-storage/pull/368#issuecomment-632960787)
> 
> We will probably test the change in both bare and expo managed apps and then merge the PR later this week.

Thank you so much, good to hear and keep up the great work!

**Comment by duongvanba** [comment]
There are any way to apply this pull request for my projet?

**Comment by theharness** [comment]
@Ashish-Nanda Any updates on timeframe?

**Comment by jvgaeta** [comment]
Any update on when this will be merged?

**Comment by Ashish-Nanda** [comment]
Hi,
Thank you for your patience. We realized the PR needs some more changes and have a bit more testing to do before we merge and release. We have prioritized this now, and aim to get it out before end of next week.

File: packages/aws-appsync-react/package.json [hunk 1]
```diff
@@ -20,13 +20,13 @@
     "prop-types": "^15.6.0"
   },
   "peerDependencies": {
-    "@react-native-community/netinfo": "4.x.x",
+    "@react-native-community/netinfo": "^5.0.0",
```
Comment (Ashish-Nanda) [inline packages/aws-appsync-react/package.json:23]: This should probably be moved to peer dependencies in the `aws-appsync` package

File: packages/aws-appsync/package.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
     "test-watch": "jest --watch"
   },
   "dependencies": {
-    "@redux-offline/redux-offline": "2.5.2-native.0",
+    "@redux-offline/redux-offline": "2.5.2-native.3",
```
Comment (Ashish-Nanda) [inline packages/aws-appsync/package.json:22]: We should add '@react-native-community/async-storage' to peer dependencies and docs as well.

**Comment by amhinson** [comment]
Tested apps generated with RN CLI & Expo with Verdaccio, and all seems to be working as expected 👍

**Comment by Ashish-Nanda** [comment]
> Tested apps generated with RN CLI & Expo with Verdaccio, and all seems to be working as expected 👍

Thanks for verifying @amhinson!

**Comment by mikeRChambers610** [comment]
I don't understand, what is the fix for the issue? NPM install new aws packages? Which ones?

**Comment by theharness** [comment]
@mikeRChambers610 Just install the latest aws-app sync package and you’ll have the fix.

**Pull request merged**